molgenis/molgenis-emx2
# fix(sql): speed up paging when using limit/offset

closes #3666

motivated by #3481 
=> query is faster but I think client side rendering is the bottleneck actually here.

What are the main changes you did:
- remove fail fast; you want to see al failing tests
- filter subquery is not added when empty
- limit/offset/orderby is applied in subquery so json is not rendered for all data before limit/offset/orderby is applied

how to test:
- look at the sql (uncomment line 289 in SqlQuery)
- test #3481 <= use a download from catalogue-acc otherwise you will not see the difference.
- take long table and go to last page, e.g. http://localhost:8080/catalogue/tables/#/VariableMappings?_page=5360
- in particular bad when sorting.

Todo:
- [ ] doesn't work well for the nested queries

N.B. I have uploaded this data but you might need to do it again if preview has been refreshed
